sympy/vector/tests/test_dyadic.py

```diff
@@ -62,6 +62,7 @@ def test_dyadic():
     q = symbols('q')
     B = A.orient_new_axis('B', q, A.k)
     assert express(d1, B) == express(d1, B, B)
+    # This assertion fails when running with the cache off:
     assert express(d1, B) == ((cos(q)**2) * (B.i | B.i) + (-sin(q) * cos(q)) *
             (B.i | B.j) + (-sin(q) * cos(q)) * (B.j | B.i) + (sin(q)**2) *
             (B.j | B.j))
```

Comment:
This can probably be changed to `(express(d1, B) - expr).simplify() == 0` that might be sufficient.

Reply:
Do you know why it would fail without the cache? This probably hints at a problem somewhere.

You can test this with
```console
$ SYMPY_USE_CACHE=no pytest sympy/vector/tests/test_dyadic.py -k test_dyadic
```
The assertion fails with
```python
assert (cos(q)**2)*(B.i|B.i) + (-sin(q)*cos(q))*(B.i|B.j) + (-sin(q)*cos(q))*(B.j|B.i) + (sin(q)**2)*(B.j|B.j)
    == (((((cos(q) ** 2) * (B.i | B.i)) + ((-sin(q) * cos(q)) * (B.i | B.j))) + ((-sin(q) * cos(q)) * (B.j | B.i))) + ((sin(q) ** 2) * (B.j | B.j)))
```